tests/cross_functional/system_test/test_clone_deletion_after_max_cluster_space_utilization.py

```diff
@@ -0,0 +1,214 @@
+    ):
+        """
+        Steps:
+            1. Have a cluster with OCP + ODF internal mode
+            2. Create cephfs + RBD pvcs with data
+            3. Create clones of pvcs till cluster full ratio.
+            4. Make the cluster out of full by increasing the full ratio.
+            5. After the cluster is out of full state and IOs started , Try to delete clones.
+            6. Clone deletion should be successful and should not give error messages.
+        """
+        # Creating the clones one by one and wait until they bound
+        self.timeout = 1800
+        logger.info(
+            f"Start creating {self.num_of_clones} clones on {interface_type} PVC of size {self.pvc_size} GB."
+        )
+        clones_list = []
+
+        if interface_type == constants.CEPHBLOCKPOOL:
+            for clone_num in range(self.num_of_clones + 1):
+                logger.info(f"Start creation of clone number {clone_num}.")
+                cloned_pvc_obj = pvc_clone_factory(
+                    self.pvc_obj, storageclass=self.pvc_obj.backed_sc, timeout=600
+                )
+                cloned_pvc_obj.reload()
+
+                # flatten the image
+                self.flatten_image(cloned_pvc_obj)
+
+                clones_list.append(cloned_pvc_obj)
+                logger.info(
+                    f"Clone with name {cloned_pvc_obj.name} for {self.pvc_size} pvc {self.pvc_obj.name} was created."
+                )
+
+        else:
+            for clone_num in range(self.num_of_clones + 1):
+                logger.info(f"Start creation of clone number {clone_num}.")
+                cloned_pvc_obj = pvc_clone_factory(
+                    self.pvc_obj, storageclass=self.pvc_obj.backed_sc, timeout=600
+                )
+                cloned_pvc_obj.reload()
+                clones_list.append(cloned_pvc_obj)
+                logger.info(
+                    f"Clone with name {cloned_pvc_obj.name} for {self.pvc_size} pvc {self.pvc_obj.name} was created."
+                )
+
+        logger.info(
+            "Verify 'CephClusterCriticallyFull' ,CephOSDNearFull Alerts are seen "
+        )
+        expected_alerts = ["CephOSDNearFull", "CephOSDCriticallyFull"]
+        prometheus = PrometheusAPI(threading_lock=threading_lock)
+        sample = TimeoutSampler(
+            timeout=600,
+            sleep=10,
+            func=prometheus.verify_alerts_via_prometheus,
+            expected_alerts=expected_alerts,
+            threading_lock=threading_lock,
+        )
+        if not sample.wait_for_func_status(result=True):
+            logger.error(f"The alerts {expected_alerts} do not exist after 1200 sec")
+            raise TimeoutExpiredError
+
+        # Make the cluster out of full by increasing the full ratio.
+        logger.info("Change Ceph full_ratio from from 85% to 95%")
+
+        change_ceph_full_ratio(95)
+        # After the cluster is out of full state and IOs started , Try to delete clones.
```

Comment:
where are we starting the IOs? I see you mentioned IOs started in the above comment

Reply:
yeah.  Its wrong info we are not starting Ios